### ISSUE TITLE
[Discussion] PEP-8 compliance

### ISSUE THREAD

**Issue by CRImier** [ZeroPhone/ZPUI#43]
Following #24 (and [CRImier/pyLCI#1](https://github.com/CRImier/pyLCI/issues/1), it'd be great to have PEP-8 and PEP-257 compliance. My ideas about this:

1. Bring the current code into compliance
2. Add instructions for contributors so that they can automatically check their contributions
3. Research (if available, add) automated hooks for checking (and/or enforcing) PEP-8 compliance

**Comment by j340m3** [comment]
You could check [flake8](http://flake8.pycqa.org/en/latest/) for PEP-8 compliance. I integrated it also to my travis-ci build. It works flawlessly. 

I've never checked for PEP-257 compliance, but there seems to be [something](https://github.com/PyCQA/pydocstyle) that does the job.

**Comment by j340m3** [comment]
I used tox. So mine is way more complicated. But here you are: [.travis.yml](https://github.com/j340m3/python-msquaredc/blob/master/.travis.yml)

**Comment by CRImier** [comment]
I see, flake8 is covered by tox, then. BTW, in your .travis.yml, why is env->global->secure (which looks like something *secure* =D ) not in Travis env vars?

**Comment by j340m3** [comment]
Yes, it is my encrypted pypi password, so that travis can upload on success to pypi without me having to type in the password. 

I followed a manual, and they have put it there, and it works. I fear touching the configuration again for such minor things 🙈.

**Comment by CRImier** [comment]
=D Got it! I sure hope nobody can use this secret to upload on your behalf, lest we get more [PyPi malware](https://www.bleepingcomputer.com/news/security/ten-malicious-libraries-found-on-pypi-python-package-index/) - so if that's the case, consider regenerating it and sticking it into the envvars =D

**Comment by j340m3** [comment]
If they are able to merge into main without my permission, they can upload, otherwise you need to fake the webhook of the push and you need to guess my private key stored in the github server. So it is possible but very unlikely.

Actually how I did it in my repo is how most projects with automated pypi upload handle it.

**Comment by CRImier** [comment]
I see, makes sense. Let's be on the look out for any GH exploits, then =D
